capsules/src/qdec.rs

```diff
@@ -0,0 +1,108 @@
+//! Provides userspace access to the Qdec on a board.
```

Comment:
Can you expand on this a bit? For one, I had to google `QDec` as it's not a shorthand for Quadrature Decoder that I'd seen before.

Is the intent of this capsule exclusively to provide an interface for hardware decoders? This is something that's fairly easy (although possibly compute-intensive) to emulate in software. I think the driver interface here probably can/should be abstract over the type of underlying implementation?

Reply:
Thanks for the comment! I will update the comment to include a brief summary of what a QDEC is, as well as a note that Quadrature Decoder is the full name.  

As for the second part of your comment, this capsule does provide an interface for hardware decoders (specifically those that decode quadrature encoded signals), so it should be abstract over the underlying implementations/chips it is applied to. Are there sections of the capsule that look too nRF specific?